src/main/java/io/javalin/core/compression/BrotliHttpOutputInterceptor.java

```diff
@@ -0,0 +1,225 @@
+package io.javalin.core.compression;
+
+import org.eclipse.jetty.http.HttpField;
+import org.eclipse.jetty.http.HttpFields;
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.http.MimeTypes;
+import org.eclipse.jetty.server.HttpChannel;
+import org.eclipse.jetty.server.HttpOutput;
+import org.eclipse.jetty.server.Response;
+import org.eclipse.jetty.util.Callback;
+import org.eclipse.jetty.util.StringUtil;
+import org.eclipse.jetty.util.log.Log;
+import org.eclipse.jetty.util.log.Logger;
+
+import java.nio.ByteBuffer;
+import java.nio.channels.WritePendingException;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.zip.CRC32;
+
+import static org.eclipse.jetty.http.CompressedContentFormat.BR;
+
+/**
+ * Majority of this code is copied from GzipHttpOutputInterceptor
+ */
+public class BrotliHttpOutputInterceptor implements HttpOutput.Interceptor {
```

Comment:
This should be converted to Kotlin. If you're using IntelliJ there's a shortcut for converting it (CTRL+SHIFT+ALT+K, I think).

Also, the whole file is very low level. When you wrote about this in the issue, I thought you were going to do something like:


```kotlin
class BrotliAndGzipInterceptor : GzipInterceptor {

    override fun intercept() {
        // code here
    }

}
```

And that would that. I guess I was a bit too optimistic? 😬